pennylane/ops/op_math/condition.py

```diff
@@ -241,14 +256,16 @@ def __call_capture_enabled(self, *args, **kwargs):
                 jaxpr_branches.append(None)
                 consts_slices.append(slice(0, 0))
             else:
-                jaxpr = jax.make_jaxpr(
-                    functools.partial(fn, **kwargs), abstracted_axes=abstracted_axes
-                )(*args)
+                f = FlatFn(functools.partial(fn, **kwargs))
+                if jax.config.jax_dynamic_shapes:
+                    f = _all_output_shapes(f)
+                jaxpr = jax.make_jaxpr(f, abstracted_axes=abstracted_axes)(*args)
                 jaxpr_branches.append(jaxpr.jaxpr)
                 consts_slices.append(slice(end_const_ind, end_const_ind + len(jaxpr.consts)))
                 consts += jaxpr.consts
                 end_const_ind += len(jaxpr.consts)
 
+        _validate_jaxpr_returns(jaxpr_branches)

```

Comment:
Very happy to upstream this validation out of `def_abstract_eval`. It will be much more robust to validate this as early as possible.